File: optimize_images.py
import os
from PIL import Image

def convert_to_webp(source_path, dest_path, quality=80):
    if os.path.exists(source_path):
        print(f"Compressing {source_path}...")
        img = Image.open(source_path)
        img.save(dest_path, "WEBP", quality=quality)
        orig_size = os.path.getsize(source_path)
        new_size = os.path.getsize(dest_path)
        print(f"Done! Saved to {dest_path}")
        print(f"Original size: {orig_size / (1024*1024):.2f} MB | Optimized size: {new_size / (1024*1024):.2f} MB ({100 - (new_size/orig_size)*100:.1f}% reduction)\n")
    else:
        print(f"Source not found: {source_path}")

if __name__ == "__main__":
    # Contact page images
    convert_to_webp("public/images/contact/tree.png", "public/images/contact/tree.webp")
    convert_to_webp("public/images/contact/grass.png", "public/images/contact/grass.webp")
    
    # About page images
    convert_to_webp("public/images/about/about_top.png", "public/images/about/about_top.webp")

File: optimize_images.js
const sharp = require("sharp");
const fs = require("fs");
const path = require("path");

async function convertToWebp(sourcePath, destPath, quality = 80) {
  if (fs.existsSync(sourcePath)) {
    console.log(`Compressing ${sourcePath}...`);
    const origSize = fs.statSync(sourcePath).size;
    
    try {
      const info = await sharp(sourcePath)
        .webp({ quality })
        .toFile(destPath);
      
      const newSize = info.size;
      console.log(`Done! Saved to ${destPath}`);
      console.log(`Original size: ${(origSize / (1024*1024)).toFixed(2)} MB | Optimized size: ${(newSize / (1024*1024)).toFixed(2)} MB (${(100 - (newSize/origSize)*100).toFixed(1)}% reduction)\n`);
    } catch (err) {
      console.error(`Error compressing ${sourcePath}:`, err);
    }
  } else {
    console.log(`Source not found: ${sourcePath}`);
  }
}

async function main() {
  // Contact page images
  await convertToWebp("public/images/contact/tree.png", "public/images/contact/tree.webp");
  await convertToWebp("public/images/contact/grass.png", "public/images/contact/grass.webp");
  
  // About page images
  await convertToWebp("public/images/about/about_top.png", "public/images/about/about_top.webp");

  // Home page images
  await convertToWebp("public/images/home/image1.png", "public/images/home/image1.webp");
  await convertToWebp("public/images/home/image2.png", "public/images/home/image2.webp");
  await convertToWebp("public/images/home/image5.png", "public/images/home/image5.webp");
  await convertToWebp("public/images/home/image6.png", "public/images/home/image6.webp");
  await convertToWebp("public/images/home/image7.png", "public/images/home/image7.webp");
  await convertToWebp("public/images/home/image8.png", "public/images/home/image8.webp");
  await convertToWebp("public/images/home/image9.png", "public/images/home/image9.webp");

  // Hero page cloud boundary
  await convertToWebp("public/clouds.png", "public/clouds.webp");
}

main();
